keps/sig-node/4800-cpumanager-split-l3cache/README.md

```diff
@@ -0,0 +1,1235 @@
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+##### Unit tests
+
+We plan on adding/modifying functions to the following files to create the uncorecache alignment feature:
+- pkg/kubelet/cm/cpumanager/cpu_assignment.go
+- /pkg/kubelet/cm/cpumanager/cpu_manager.go
+- /pkg/kubelet/cm/cpumanager/policy_options.go
+- /pkg/kubelet/cm/cpumanager/topology/topology.go
+
+Existing topoology test cases will be modified to include uncorecache topology. All modified and added functions will have new test cases.
+
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+- `/pkg/kubelet/cm/cpumanager/cpu_assignment.go`: `2024-08-26` - `93.5%`
+- `/pkg/kubelet/cm/cpumanager/cpu_manager.go`: `2024-08-26` - `74.5%`
+- `/pkg/kubelet/cm/cpumanager/policy_static.go`: `2024-08-26` - `89.5%`
+
+##### Integration tests
+
+<!--
+Integration tests are contained in k8s.io/kubernetes/test/integration.
+Integration tests allow control of the configuration parameters used to start the binaries under test.
+This is different from e2e tests which do not allow configuration of parameters.
+Doing this allows testing non-default options and multiple different and potentially conflicting command line options.
+-->
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+Will follow existing integration testing required in PR check.
+
+##### e2e tests
+
+- e2e testing should cover the deployment of a pod that is following LLC alignment which can be verified in the CPU Manager State file
```

Comment:
If they can't check the state file then I'm not sure what else they could use besides some external tool such as top.

Reply:
checking the state file should be really kept as last resort. The good news is in the context of e2e_node tests, checking through the podresources API is not hard, we have plenty of examples already

Reply:
Do you know if CacheIDs are already reported in the pod resource API?

Reply:
they are not, but I don't think they are exposed in the `cpu_manager_state` file either. Should not be a big problem for e2e_node tests though. The e2e_node tests are meant to run in the same machine kubelet is running, by design. And they require root privileges anyway (to run the kubelet). So we can learn the cpu assignment from podresources API, and learn beck the uncore cache id from sysfs in the e2e_node tests.
It's not straightforward as it could be, but seems still doable, and I still strongly prefer this approach over reading the cpu_manager_state file.

Reply:
updated. will use podresources API to determine cpu assignments and cross-reference to sysfs topology